### PR TITLE
Add builtin modules with node: prefix

### LINKED ISSUE #13290: New node: scheme from Node 16 not handled
# Bug report

**What is the current behavior?**

The [new `node:` prefix from Node v16](https://twitter.com/jasnell/status/1385238402149150721) causes an `UnhandledSchemeError`:

```
ERROR in node:path
Module build failed: UnhandledSchemeError: Reading from "node:path" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
    at /Users/k/p/webpack-babel-test/node_modules/webpack/lib/NormalModule.js:742:26
    at Hook.eval [as callAsync] (eval at create (/Users/k/p/webpack-babel-test/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/k/p/webpack-babel-test/node_modules/tapable/lib/Hook.js:18:14)
    at Object.processResource (/Users/k/p/webpack-babel-test/node_modules/webpack/lib/NormalModule.js:739:9)
    at processResource (/Users/k/p/webpack-babel-test/node_modules/loader-runner/lib/LoaderRunner.js:220:11)
    at iteratePitchingLoaders (/Users/k/p/webpack-babel-test/node_modules/loader-runner/lib/LoaderRunner.js:171:10)
    at runLoaders (/Users/k/p/webpack-babel-test/node_modules/loader-runner/lib/LoaderRunner.js:397:2)
    at NormalModule.doBuild (/Users/k/p/webpack-babel-test/node_modules/webpack/lib/NormalModule.js:729:3)
    at NormalModule.build (/Users/k/p/webpack-babel-test/node_modules/webpack/lib/NormalModule.js:878:15)
    at /Users/k/p/webpack-babel-test/node_modules/webpack/lib/Compilation.js:1311:12
 @ ./src/index.tsx 1:0-29 2:12-16

webpack 5.36.0 compiled with 1 error in 977 ms
```

This [will be backported to at least also Node v14](https://twitter.com/jasnell/status/1385239259838222336)

**If the current behavior is a bug, please provide the steps to reproduce.**

Minimal reproduction repo (run `yarn start`): https://github.com/karlhorky/webpack-node-16-schema

**What is the expected behavior?**

webpack should throw a "Module not found" error with the polyfills warning, just like when you `import path from 'path'`:

```
ERROR in ./src/index.tsx 1:0-24
Module not found: Error: Can't resolve 'path' in '/Users/k/p/webpack-babel-test/src'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }
resolve 'path' in '/Users/k/p/webpack-babel-test/src'
  Parsed request is a module
  using description file: /Users/k/p/webpack-babel-test/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /Users/k/p/webpack-babel-test/src/node_modules doesn't exist or is not a directory
      looking for modules in /Users/k/p/webpack-babel-test/node_modules
        single file module
          using description file: /Users/k/p/webpack-babel-test/package.json (relative path: ./node_modules/path)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/k/p/webpack-babel-test/node_modules/path doesn't exist
            *
              Field 'browser' doesn't contain a valid alias configuration
              /Users/k/p/webpack-babel-test/node_modules/path* doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/k/p/webpack-babel-test/node_modules/path.js doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/k/p/webpack-babel-test/node_modules/path.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/k/p/webpack-babel-test/node_modules/path.tsx doesn't exist
        /Users/k/p/webpack-babel-test/node_modules/path doesn't exist
      /Users/k/p/node_modules doesn't exist or is not a directory
      /Users/k/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory

webpack 5.36.0 compiled with 1 error in 913 ms
```

**Other relevant information:**
webpack version: 5.36.2
Node.js version: 16.0.0
Operating System: macOS Big Sur 11.3
Additional tools:

### PR DESCRIPTION
**Pull request by karlhorky**
Closes #13290 

**What kind of change does this PR introduce?**

Support `node:` prefix for builtin modules

**Did you add tests for your changes?**

@alexander-akait what tests would you suggest here?

**Does this PR introduce a breaking change?**

I guess not, since this is now a feature of Node.js?

**What needs to be documented once your changes are merged?**

Not sure this needs to be documented, but happy to be corrected here.

### CODE REVIEW

**Review by sokra — CHANGES_REQUESTED** [review]
there is already a PR for that: https://github.com/webpack/webpack/pull/12693

**Comment by karlhorky** [comment]
@sokra Ok thanks for the hint. Can you provide some guidance as to what a test case would look like? And where I would add this test case?

**Comment by alexander-akait** [comment]
@karlhorky You can put this here https://github.com/webpack/webpack/tree/master/test/configCases/target and add `require`/`import` with `node` prefix

**Comment by karlhorky** [comment]
@alexander-akait ok I spent 10 minutes looking at the things in that folder, but I don't know what you mean.

I ended up just guessing that you mean doing something like this: [`12108b3` (#13311)](https://github.com/webpack/webpack/pull/13311/commits/12108b3655e77b8d23de1fc714faab9fd497ae9c)

File: test/configCases/target/amd-unnamed/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 it("should run", function() {});
 
 it("should name define", function() {
-	var fs = require("fs");
+	var fs = require("node:fs");
```
Comment (alexander-akait) [inline test/configCases/target/amd-unnamed/index.js:4]: Please add new tests, don't modify exists

Comment (karlhorky) [reply test/configCases/target/amd-unnamed/index.js:4]: As I wrote above, I don't understand how to add new tests here - I tried :)

**Comment by karlhorky** [comment]
So this feature is about to get more usage: the `node:` scheme / prefix / protocol was backported to Node.js 14.13.1 / 12.20.0 🎉

https://nodejs.org/api/esm.html -> link "node: Imports" (links with anchors don't scroll properly right now)

Screen Shot 2021-05-30 at 11 04 39

**Comment by karlhorky** [comment]
An alternative to the approaches in #13311 and #12693 would be to use [`is-core-module`](https://github.com/inspect-js/is-core-module) (which was split from [`resolve`](https://github.com/browserify/resolve) in [browserify/resolve@7c26483](https://github.com/browserify/resolve/commit/7c26483)), so that this list doesn't need to be maintained and tested by `webpack` itself.

**Comment by alexander-akait** [comment]
Done here https://github.com/webpack/webpack/pull/12693, sorry for delay

**Comment by karlhorky** [comment]
No worries, thanks for this!

**Pull request closed without merging**